### PR TITLE
Introducing Automated Documentation Tag Updates

### PR DESCRIPTION
**Pull request by ImmaZoni**
**Overview:**
This pull request introduces a new automated release workflow for our documentation site. In the past, we relied on manual updates to keep our documentation tags current. With this workflow, we can now automatically track and update the latest version tags for Pulsar and Advanced CLI in our documentation, ensuring that our users always have access to the most up-to-date information.

**Changes:**
- Added a scheduled GitHub Action workflow that checks for new releases in the Pulsar and Advanced CLI repositories.
- Implemented logic to update the documentation tags only when a new release is marked as "latest" and not a prerelease.
- Utilized whitelists and blacklists to filter out irrelevant releases and ensure accuracy.
- Created a new branch and pull request to apply the updates to the documentation code.

**Why is this important?**
This automated release workflow streamlines the process of keeping our documentation up to date, reducing the risk of human error and ensuring that our users always have access to accurate version information. It enhances the overall user experience and demonstrates our commitment to providing high-quality documentation.

**Additional Notes:**
This is our first step towards automating our release process. For the time being I have it set to create a PR, we will set it to auto merge the PR once we have verified it works correctly. Feedback and suggestions are welcome as we continue to refine and expand our automation efforts.

### CODE REVIEW

File: .github/workflows/automated-release.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Update documentation code
+        id: update-documentation-code
+        run: |
+          current_pulsar_tag=${{ env.current_pulsar_tag }}
+          current_substrate_cli_tag=${{ env.current_substrate_cli_tag }}
+          latest_pulsar_tag=${{ steps.pulsar_release.outputs.pulsar_tag }}
+          latest_advanced_cli_tag=${{ steps.advanced_cli_release.outputs.advanced_cli_tag }}
+
+          # Define arrays for whitelists and blacklists for each repository, each variable can hold an array such as ("first_word" "second_word")
+          pulsar_whitelist=("targetted-word")
```
Comment (EmilFattakhov) [inline .github/workflows/automated-release.yml:49]: What would be the whitelist `"targetted-word"` for the pulsar in our case? I don't see any patters we can define it on, they're all called `alpha`.

File: .github/workflows/automated-release.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          pulsar_blacklist=("unwanted-word")
```
Comment (EmilFattakhov) [inline .github/workflows/automated-release.yml:50]: And not sure about the blacklist words too, do you have any ideas?

**Review by EmilFattakhov — APPROVED** [review]
This looks awesome Justin, I like the approach! 
Left a few non-blocking questions, though I believe the update won't go through without setting them properly.

File: .github/workflows/automated-release.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      # Compare the latest release tags with the current versions and update the documentation code
+      - name: Update documentation code
+        id: update-documentation-code
+        run: |
+          current_pulsar_tag=${{ env.current_pulsar_tag }}
+          current_substrate_cli_tag=${{ env.current_substrate_cli_tag }}
+          latest_pulsar_tag=${{ steps.pulsar_release.outputs.pulsar_tag }}
+          latest_advanced_cli_tag=${{ steps.advanced_cli_release.outputs.advanced_cli_tag }}
+
+          # Define arrays for whitelists and blacklists for each repository, each variable can hold an array such as ("first_word" "second_word")
```
Comment (jim-counter) [inline .github/workflows/automated-release.yml:48]: Consider allowlist and denylist over whitelist and blacklist.

File: .github/workflows/automated-release.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          pulsar_whitelist=("targetted-word")
+          pulsar_blacklist=("unwanted-word")
+          advanced_cli_whitelist=("gemini-3f")
+          advanced_cli_blacklist=("runtime" "snapshot")
+
+          # Check if the latest release should be updated based on whitelists and blacklists
```
Comment (jim-counter) [inline .github/workflows/automated-release.yml:54]: Consider a regular expression approach to do this. Maybe something for next time you're faced with a challenge that requires string matching.

**Review by jim-counter — APPROVED** [review]
Approved but a _suggested_ change and a hint for a potentially more efficient way to handle this kind of problem in the future.

File: .github/workflows/automated-release.yml (hunk 1 above)
Comment (ImmaZoni) [reply .github/workflows/automated-release.yml]: Removed the filler words, while its not needed for pulsar now, I figured that the option may come in handy when pulsar starts getting stable and pre-releases.

File: .github/workflows/automated-release.yml (hunk 3 above)
Comment (ImmaZoni) [reply .github/workflows/automated-release.yml]: Updated to reflect your suggestion 😄

**Pull request merged**